src/sentry/rules/processing/delayed_processing.py

```diff
@@ -92,42 +100,75 @@ def get_rules_to_slow_conditions(
     return rules_to_slow_conditions
 
 
-def get_condition_groups(
+def _generate_unique_queries(
+    condition_data: EventFrequencyConditionData, environment_id: int
+) -> list[UniqueConditionQuery]:
+    """
+    Returns a list of all unique condition queries that must be made for the
+    given condition instance.
+    Count comparison conditions will only have one unique query, while percent
+    comparison conditions will have two unique queries.
+    """
+    unique_queries = [
+        UniqueConditionQuery(
+            cls_id=condition_data["id"],
+            interval=condition_data["interval"],
+            environment_id=environment_id,
+        )
+    ]
+    if condition_data.get("comparisonType") == ComparisonType.PERCENT:
+        # We will later compare the first query results against the second query to calculate
+        # a percentage for percentage comparison conditions.
+        comparison_interval = condition_data.get("comparisonInterval", DEFAULT_COMPARISON_INTERVAL)
+        second_comparison_query = unique_queries[0]._replace(comparisonInterval=comparison_interval)
+        unique_queries.append(second_comparison_query)
+
+    return unique_queries
+
+
+def get_condition_query_groups(
     alert_rules: list[Rule], rules_to_groups: DefaultDict[int, set[int]]
-) -> dict[UniqueCondition, DataAndGroups]:
+) -> dict[UniqueConditionQuery, DataAndGroups]:
     """
-    Map unique conditions to the group IDs that need to checked for that
-    condition. We also store a pointer to that condition's JSON so we can
-    instantiate the class later
+    Map unique condition queries to the group IDs that need to checked for that
+    query. We also store a pointer to that condition's JSON so we can
+    instantiate the class later.
     """
-    condition_groups: dict[UniqueCondition, DataAndGroups] = {}
+    condition_groups: dict[UniqueConditionQuery, DataAndGroups] = {}
     for rule in alert_rules:
         # We only want a rule's slow conditions because alert_rules are only added
         # to the buffer if we've already checked their fast conditions.
         slow_conditions = get_slow_conditions(rule)
         for condition_data in slow_conditions:
-            if condition_data:
-                unique_condition = UniqueCondition(
-                    condition_data["id"], condition_data["interval"], rule.environment_id
-                )
+            for unique_cond in _generate_unique_queries(condition_data, rule.environment_id):
```

Comment:
maybe just `condition` rather than abbreviating?

Reply:
Swapped to `condition_query`